lib/puppet/environments.rb

```diff
@@ -447,5 +451,27 @@ def expired?
         Time.now > @ttl
       end
     end
+
+    class FileMtimeEntry < Entry
+      def initialize(env, path)
+        super value
+        @mtime   = Time.now
+        @evicted = false
+        @thread  = Thread.new do
```

Comment:
how many threads does this create? Can we instead compute `@evicted` when `evicted?` is called?